presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ParquetReader.java

```diff
@@ -206,7 +210,16 @@ private ColumnChunk readPrimitive(PrimitiveField field)
             ParquetColumnChunk columnChunk = new ParquetColumnChunk(descriptor, buffer, 0);
             columnReader.setPageReader(columnChunk.readAllPages());
         }
-        return columnReader.readPrimitive(field);
+        try {
+            return columnReader.readPrimitive(field);
+        }
+        catch (UnsupportedOperationException e) {
```

Comment:
`UnsupportedOperationException` is a broad exception to catch here. Maybe we should throw a specific Parquet exception and catch it here. What do you think?

Reply:
sure. UnsupportedOperationException is actually from type.writeLong etc, when lower level ColumnReader trying to read parquet values. How about throw ParquetCorruptionException there, and catch it here?